### ISSUE TITLE
Performance testing - capitation

### ISSUE THREAD

**Issue by lymychp** [edenlabllc/ehealth.api#877]
As a <>  
I want <>  
So that <>

**Comment by ybod** [comment]
https://github.com/edenlabllc/performance_testing

**Comment by AlexKovalevych** [comment]
Tested with locust on preprod with configuration:

min_weight = 10
max_weight = 10
users = 100

### GET /reports/stats
annon: 950 rps (max). 3 replicas on different physical nodes.
Uses maximum CPU (maximum of available), hangs after high load and didn't back to normal after load decreased. Only pods manual restart helped.
kong: 1500 rps(max). 3 replicas on the same physical node. Default setup used, no extra plugins enabled.
Uses ~10% of CPU (each instance), reports api microservice limit reached (so kong can do more)

### POST /api/employee_request
annon: 100 rps (max). IL api microservice limit reached (gateway can do more)
kong: not tested yet (need to create a Docker image with mithril plugin)

#### In conclusion
In synthetic tests kong average response time is 8ms faster than gateway (44 against 52), the service is stable and response time doesn't decrease after period of high load and CPU load is close to 0 without load. Kong rps is higher in 1.5 times than gateway (available resources and conditions are equal).

**Comment by AlexKovalevych** [comment]
Tested again with a single instance of gateway and kong on dev. Single instance of reports pod and two instances of il pod.

GET `/reports/stats`

Gateway: 256.17rps
Kong: 329.15rps

POST `/api/employee_requests` (create employee_request)

Gateway: 42.25rps
Kong: 49.22rps

In both cases, microservices limit reached. Gateway and kong can do more, but same difference in CPU usage. In conclusion, Kong consumes less CPU, faster (up to 1.5 times) and is more stable especially on high load.
